pyked/chemked.py

```diff
@@ -233,6 +241,238 @@ def get_dataframe(self, output_columns=None):
+        reference.set('preferredKey', citation)
+        reference.set('doi', self.reference.doi)
+
+        apparatus = etree.SubElement(root, 'apparatus')
+        kind = etree.SubElement(apparatus, 'kind')
+        kind.text = self.apparatus.kind
+
+        common_properties = etree.SubElement(root, 'commonProperties')
+        # ChemKED objects have no common properties once loaded. Check for properties
+        # among datapoints that tend to be common
+        common = []
+        composition = self.datapoints[0].composition
+        # Composition type *has* to be the same
+        composition_type = self.datapoints[0].composition_type
+        if all([composition == dp.composition for dp in self.datapoints]):
+            # initial composition is common
+            common.append('composition')
+            prop = etree.SubElement(common_properties, 'property')
+            prop.set('name', 'initial composition')
+
+            for species in composition:
+                component = etree.SubElement(prop, 'component')
+                species_link = etree.SubElement(component, 'speciesLink')
+                species_link.set('preferredKey', species['species-name'])
+                if species.get('InChI'):
+                    species_link.set('InChI', species['InChI'])
+
+                amount = etree.SubElement(component, 'amount')
+                amount.set('units', composition_type)
+                amount.text = str(species['amount'].magnitude)
+
+        # If multiple datapoints present, then find any common properties. If only
+        # one datapoint, then composition should be the only "common" property.
+        if len(self.datapoints) > 1:
+            for prop_name in datagroup_properties:
+                attribute = prop_name.replace(' ', '_')
+                quantity = getattr(self.datapoints[0], attribute)
+                if (quantity != None and
+                    all([quantity == getattr(dp, attribute) for dp in self.datapoints])
+                    ):
+                    common.append(prop_name)
+                    prop = etree.SubElement(common_properties, 'property')
+                    prop.set('description', '')
+                    prop.set('name', prop_name)
+                    prop.set('units', str(quantity.units))
+
+                    value = etree.SubElement(prop, 'value')
+                    value.text = str(quantity.magnitude)
+
+        # Ignition delay can't be common, unless only a single datapoint.
+
+        datagroup = etree.SubElement(root, 'dataGroup')
+        datagroup.set('id', 'dg1')
+        datagroup_link = etree.SubElement(datagroup, 'dataGroupLink')
+        datagroup_link.set('dataGroupID', '')
+        datagroup_link.set('dataPointID', '')
+
+        property_idx = {}
+        labels = {'temperature': 'T', 'pressure': 'P',
+                  'ignition delay': 'tau', 'pressure rise': 'dP/dt',
+                  }
+
+        for prop_name in datagroup_properties:
```

Comment:
gotcha. I updated the middle block with my other change to handle (potential) variable species sets, but otherwise looks good.